FEAT #9703: l10n_ve_invoice

AlbertHamm-binaural opened the pull request:
Problema: El boton cancelar debe aparecer disponible en las notas de debito y credito si forman parte del mismo período fiscal.

Solución: Cálculo de entry_in_period y control de visibilidad del botón de cancelar

- Se agregó un campo booleano computado `entry_in_period` en `account.move` para identificar si una nota de débito o crédito está dentro del período fiscal vigente.
- La lógica considera el tipo de contribuyente (especial, ordinario, formal) y la fecha de la factura en relación con el límite del período.
- Se actualizó la vista XML para incluir el campo y condicionar la visibilidad del botón de cancelar: solo se muestra si el documento está dentro del período fiscal y cumple con los criterios de estado y tipo.

Tarea (Link): https://binaural.odoo.com/web?db=binaural-dev-binaural-release-10413381&token=4P6sT2NxLGgos2OwxwuG#id=9703&cids=2&menu_id=302&action=386&model=helpdesk.ticket&view_type=form

Tarea de proyecto []
Ticket de soporte [x]

AlbertHamm-binaural committed:
FEAT #9703: l10n_ve_invoice

Problema: El boton cancelar debe aparecer disponible en las notas de debito y credito si forman parte del mismo período fiscal.

Solución: Cálculo de entry_in_period y control de visibilidad del botón de cancelar

- Se agregó un campo booleano computado `entry_in_period` en `account.move` para identificar si una nota de débito o crédito está dentro del período fiscal vigente.
- La lógica considera el tipo de contribuyente (especial, ordinario, formal) y la fecha de la factura en relación con el límite del período.
- Se actualizó la vista XML para incluir el campo y condicionar la visibilidad del botón de cancelar: solo se muestra si el documento está dentro del período fiscal y cumple con los criterios de estado y tipo.

Tarea (Link): https://binaural.odoo.com/web?db=binaural-dev-binaural-release-10413381&token=4P6sT2NxLGgos2OwxwuG#id=9703&cids=2&menu_id=302&action=386&model=helpdesk.ticket&view_type=form

Tarea de proyecto []
Ticket de soporte [x]

AlbertHamm-binaural committed:
manifest

AlbertHamm-binaural committed:
FIX #9703: l10n_ve_invoice, l10n_ve_stock_account

Problema: La fecha de cierre fiscal se esta calculando de una forma que no se adapta a todos los meses.

Solución: Se agrego una libreria(calendar) con la cual se puede obtener el ultimo dia del mes, con esto se modifico la forma de obtener el ultimo dia del mes actual.

Tarea (Link): https://binaural.odoo.com/web#id=9703&cids=2&menu_id=302&action=386&model=helpdesk.ticket&view_type=form

Tarea de proyecto []
Ticket de soporte [x]

AlbertHamm-binaural committed:
Merge branch '17.0' into 17.0_feat-ti_9703_show_cancel_button_if_in_fiscal_period

AlbertHamm-binaural committed:
Merge branch '17.0' into 17.0_feat-ti_9703_show_cancel_button_if_in_fiscal_period

miguel-binaural reviewed (changes requested):
Revisa las pruebas que fallaron @AlbertHamm-binaural

AlbertHamm-binaural committed:
FIX #9703: l10n_ve_invoice, l10n_ve_stock_account

Problema: Pruebas unitarias fallando en l10n_ve_stock_account y faltantes sobre los cambios realizados anteriormente, manifest

Solución:
- En las pruebas unitarias de l10n_ve_stock_account se arreglo el impuesto para que no causara error con el pais y se acomodando la forma de crear la orden de compra
- Agregando pruebas unitarias al desarrollo para que saliera el boton de cancelar solo si la factura entre en el rango del periodo fiscal
- Agregando manifest

Tarea (Link): https://binaural.odoo.com/web?db=binaural-dev-binaural-release-10413381&token=4P6sT2NxLGgos2OwxwuG#id=9703&cids=2&menu_id=302&action=386&model=helpdesk.ticket&view_type=form

Tarea de proyecto [x]
Ticket de soporte []

AlbertHamm-binaural committed:
Merge branch '17.0' into 17.0_feat-ti_9703_show_cancel_button_if_in_fiscal_period

AlbertHamm-binaural committed:
FIX #9703: l10n_ve_stock_account

Problema: problema con prueba unitaria, necesita del modulo de purchase para correr el test_02 pero no esta como dependencia en el modulo

Solución: se quito ese test para luego colocarlo en el modulo de purchase

Tarea (Link): https://binaural.odoo.com/web?db=binaural-dev-binaural-release-10413381&token=4P6sT2NxLGgos2OwxwuG#id=9703&cids=2&menu_id=302&action=386&model=helpdesk.ticket&view_type=form

Tarea de proyecto [x]
Ticket de soporte []

AlbertHamm-binaural committed:
Merge branch '17.0' into 17.0_feat-ti_9703_show_cancel_button_if_in_fiscal_period

AlbertHamm-binaural committed:
Merge branch '17.0' into 17.0_feat-ti_9703_show_cancel_button_if_in_fiscal_period

AlbertHamm-binaural committed:
FIX #9703: l10n_ve_invoice, l10n_ve_tax

- Arreglando pruebas unitarias del modulo de 'l10n_ve_invoice' acerca del periodo fiscal para mostrar boton cancelar.
- Acomodando vista del modulo de 'l10n_ve_tax', vista llamada de forma erronea:
account_joural.xml ---> account_journal.xml

Tarea (Link): https://binaural.odoo.com/web?db=binaural-dev-binaural-release-10413381&token=4P6sT2NxLGgos2OwxwuG#id=9703&cids=2&menu_id=302&action=386&model=helpdesk.ticket&view_type=form

Tarea de proyecto []
Ticket de soporte [x]
